hilti/include/ast/operators/map.h

```diff
@@ -69,13 +69,14 @@ STANDARD_OPERATOR_2x_lhs(map, IndexNonConst, Index, operator_::elementType(0), t
 BEGIN_METHOD(map, Get)
     auto signature() const {
         return Signature{.self = type::Map(type::Wildcard()),
-                         .result = operator_::elementType(0),
+                         .result =
+                             [](const std::vector<Expression>& e1, const std::vector<Expression>& e2) {
+                                 return type::Optional(operator_::elementType(0)(e1, e2).value_or(type::unknown));
+                             },
                          .id = "get",
-                         .args = {{.id = "key", .type = type::Any()},
-                                  {.id = "default", .type = type::Any(), .optional = true}},
+                         .args = {{.id = "key", .type = type::Any()}},
                          .doc = R"(
-Returns the map's element for the given key. If the key does not exist, returns
-the default value if provided; otherwise throws a runtime error.
+Returns the map's element for the given key.
```

Comment:
If we keep the change, expand a bit on behavior with non-existing elements to make it explicit.